core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -377,6 +377,12 @@ private RexNode simplifyNot(RexCall call) {
     case NOT:
       // NOT NOT x ==> x
       return simplify_(((RexCall) a).getOperands().get(0));
+    case LITERAL:
+      if (a.isAlwaysTrue()) {
+        return rexBuilder.makeLiteral(false);
+      } else if (a.isAlwaysFalse()) {
```

Comment:
Please check for `getType().getSqlTypeName()` and use `RexLiteral.booleanValue(..)`

Reply:
done for this!